plugin/plugins/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,170 @@
+			http.Error(w, "invalid request", http.StatusBadRequest)
+			return
+		}
+
+		// Check content type
+		if r.Header.Get("Content-Type") != "application/json" {
+			t.Log("invalid content type")
+			http.Error(w, "invalid content type", http.StatusBadRequest)
+			return
+		}
+
+		// Check if the body is not empty
+		if r.Body == nil {
+			t.Log("empty body")
+			http.Error(w, "empty body", http.StatusBadRequest)
+			return
+		}
+
+		// Read the body
+		body, _ := io.ReadAll(r.Body)
+		if len(body) == 0 {
+			t.Log("zero-length body")
+			http.Error(w, "empty body", http.StatusBadRequest)
+			return
+		}
+
+		t.Logf("Received telemetry:\n %s", string(body))
+
+		err := json.Unmarshal(body, &e)
+		if err != nil {
+			t.Log("error unmarshaling event", err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		// Return success
+		w.WriteHeader(http.StatusOK)
+	})), func() LogEvent { return e }
+}
+
+func makeNode(t *testing.T) (node *core.IpfsNode, repopath string) {
+	t.Helper()
+
+	// Create a Temporary Repo
+	repoPath, err := os.MkdirTemp("", "ipfs-shell")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pebbledspli := pebbleds.Plugins[0]
+	pebbledspl, ok := pebbledspli.(plugin.PluginDatastore)
+	if !ok {
+		t.Fatal("bad datastore plugin")
+	}
+
+	err = fsrepo.AddDatastoreConfigHandler(pebbledspl.DatastoreTypeName(), pebbledspl.DatastoreConfigParser())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a config with default options and a 2048 bit key
+	cfg, err := config.Init(io.Discard, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cfg.Datastore.Spec = map[string]interface{}{
+		"type":               "pebbleds",
+		"prefix":             "pebble.datastore",
+		"path":               "pebbleds",
+		"formatMajorVersion": int(pebble.FormatNewest),
+	}
+
+	// Create the repo with the config
+	err = fsrepo.Init(repoPath, cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Construct the node
+
+	nodeOptions := &core.BuildCfg{
+		Online:  true,
+		Routing: libp2p.NilRouterOption,
+		Repo:    repo,
+	}
+
+	node, err = core.NewNode(context.Background(), nodeOptions)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	node.IsDaemon = true
+	return
+}
+
+func TestSendTelemetry(t *testing.T) {
+	if err := logging.SetLogLevel("telemetry", "DEBUG"); err != nil {
+		t.Fatal(err)
+	}
+	ts, eventGetter := mockServer(t)
+	defer ts.Close()
+
+	node, repoPath := makeNode(t)
+
+	// Create a plugin instance
+	p := &telemetryPlugin{
+		runOnce: true,
+	}
+
+	// Initialize the plugin
+	pe := &plugin.Environment{
+		Repo:   repoPath,
+		Config: nil,
+	}
+	err := p.Init(pe)
+	if err != nil {
+		t.Fatalf("Init() failed: %v", err)
+	}
+
+	p.endpoint = ts.URL
+
+	// Start the plugin
+	err = p.Start(node)
+	if err != nil {
+		t.Fatalf("Start() failed: %v", err)
+	}
+
+	e := eventGetter()
+	if e.UUID != p.event.UUID {
+		t.Fatal("uuid mismatch")
+	}
+}
```

Comment:
Consider using [github.com/stretchr/testify/require](https://pkg.go.dev/github.com/stretchr/testify/require) for testing.

Reply:
I like things vanilla